File: app/src/main/java/com/irishinterest/irishinterest/model/user/UserAddRemoveFavoriteBookEncrypted.java
package com.irishinterest.irishinterest.model.user;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "user_id",
        "book_id"
})
public class UserAddRemoveFavoriteBookEncrypted {

    @JsonProperty("user_id")
    private Integer userId;
    @JsonProperty("book_id")
    private Integer bookId;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("user_id")
    public Integer getUserId() {
        return userId;
    }

    @JsonProperty("user_id")
    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    @JsonProperty("book_id")
    public Integer getBookId() {
        return bookId;
    }

    @JsonProperty("book_id")
    public void setBookId(Integer bookId) {
        this.bookId = bookId;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/java/com/irishinterest/irishinterest/model/user/UserAddRemoveFavoriteBookEncrypted.kt
package com.irishinterest.irishinterest.model.user

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter
import java.util.HashMap

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("user_id", "book_id")
class UserAddRemoveFavoriteBookEncrypted {
    @get:JsonProperty("user_id")
    @set:JsonProperty("user_id")
    @JsonProperty("user_id")
    var userId: Int? = null

    @get:JsonProperty("book_id")
    @set:JsonProperty("book_id")
    @JsonProperty("book_id")
    var bookId: Int? = null

    @JsonIgnore
    private val additionalProperties: MutableMap<String, Any> = HashMap()
    @JsonAnyGetter
    fun getAdditionalProperties(): Map<String, Any> {
        return additionalProperties
    }

    @JsonAnySetter
    fun setAdditionalProperty(name: String, value: Any) {
        additionalProperties[name] = value
    }
}